SpatialGDK/Source/SpatialGDK/Public/SpatialView/CommandResponse.h

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Templates/UniquePtr.h"
+#include <improbable/c_worker.h>
+#include <improbable/c_schema.h>
+
+namespace SpatialGDK
+{
+
+struct CommandResponseDeleter
+{
+	void operator()(Schema_CommandResponse* CommandResponse) const noexcept
+	{
+		if (CommandResponse)
+		{
+			Schema_DestroyCommandResponse(CommandResponse);
+		}
+	}
+};
+
+using OwningCommandResponsePtr = TUniquePtr<Schema_CommandResponse, CommandResponseDeleter>;
+
+// An RAII wrapper for component data.
+class CommandResponse
+{
+public:
+	// Creates a new component data.
+	explicit CommandResponse(Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
+	// Takes ownership of component data.
+	explicit CommandResponse(OwningCommandResponsePtr Data, Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
+
+	~CommandResponse() = default;
+
+	// Moveable, not copyable.
+	CommandResponse(const CommandResponse&) = delete;
+	CommandResponse(CommandResponse&&) = default;
+	CommandResponse& operator=(const CommandResponse&) = delete;
+	CommandResponse& operator=(CommandResponse&&) = default;
+
+	static CommandResponse CreateCopy(const Schema_CommandResponse* Data, Worker_ComponentId ComponentId, Worker_CommandIndex CommandIndex);
```

Comment:
CreateCopy implies copying a command response, instead of the schema contents itself. Why isn't this just another CommandResponse constructor?

Reply:
Originally it was, this was a compromise due to concerns that it would make the constructor look like it was assuming ownership instead of copying, although I'm happy to just make it a constructor.

Reply:
I'm not sure I understand what copying a response is without copying the scheme contents though.

Reply:
@m-samiec would you prefer `CreateDeepCopy`?

Reply:
Personally I'd make it a constructor (and I do in my own stuff). With a const* type I'd argue what it does is fairly unambiguous. But for the same reason given this is a container type that owns a resource there is only one thing I'd expect such a static function to do.

Reply:
> I'm not sure I understand what copying a response is without copying the scheme contents though.

Sorry I wasn't clear. I meant it sounded like it should be copying a `CommandResponse` obj, instead of something with a schema root. Perhaps CreateFromSchema?

nit though - not hugely fussed.